src/components/MailboxPicker.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+	<Treeselect
+		ref="Treeselect"
+		:options="mailboxes"
+		:multiple="false"
+		:clearable="false"
+		v-bind="$attrs"
+		v-on="$listeners" />
+</template>
+<script>
+import Treeselect from '@riophae/vue-treeselect'
+import '@riophae/vue-treeselect/dist/vue-treeselect.css'
+
+export default {
+	name: 'MailboxPicker',
+	components: {
+		Treeselect,
+	},
+	props: {
+		accountid: {
+			type: Number,
+			required: true,
+		},
+	},
+	computed: {
+		mailboxes() {
+			const boxes = []
+			this.getMailboxes(boxes)
+			return boxes
+		},
+	},
+	methods: {
+		getMailboxes(root, folderid) {
+			let folders = []
+			if (!folderid) {
+				folders = this.$store.getters.getMailboxes(this.accountid)
+			} else {
+				folders = this.$store.getters.getSubMailboxes(folderid)
+			}
+			folders.forEach((folder) => {
+				const mailbox = {
+					id: folder.databaseId,
+					label: folder.displayName,
+				}
+				if (folder.mailboxes.length > 0) {
+					mailbox.children = []
+					this.getMailboxes(mailbox.children, mailbox.id)
```

Comment:
here we'll combine the result from the recursive call, like `mailboxes = mailboxes.concat(this.getMailboxes(mailbox.databaseId))`.

Reply:
We need a tree structure, not a plain array.

Reply:
Right, I missed the part with `.children`.

But still, can't you do

```
mailboxes.push({
  id: mailbox.databaseId,
  label: mailbox.displayName,
  children: this.getMailboxes(mailbox.databaseId),
})

Reply:
Ok I did it this way but it's no difference in functionality.

Reply:
I didn't claim that. It's just a different style: functional.